### PR TITLE
Refactor `test-records.ts` to prepare for reviewdog 

### PR DESCRIPTION
**Pull request by baltpeter**
Preview of what this will look like: https://github.com/einfrank/data/pull/9

image

### CODE REVIEW

File: src/checks/company/tested-requirements.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const check: CompanyCheck = {
+    id: 'tested-requirements',
+    desc: 'A `quality` of `tested` may only be set if `required-elements` are specified.',
+    url: 'https://github.com/datenanfragen/data/issues/811',
+    severity: 'ERROR',
+    run: (json) => {
+        if (json['quality'] === 'tested' && !json['required-elements']) {
+            return {
+                message:
+                    "Record has `quality` of `tested` but doesn't specify `required-elements`.\n\nPreferably set `required-elements`, otherwise demote to `verified`.",
```
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/checks/company/tested-requirements.ts:12]: I'll do that in a separate PR.

**Comment by baltpeter** [comment]
> If an address triggers both trim strings and trim address two different suggestions are generated:

Yes, I know. But I think the benefits of the simpler and clearer semantics of specifying each test separately and independent of the other ones, outweigh this downside. I don't see a way to fix this without a ridiculous increase in complexity.

**Comment by baltpeter** [comment]
> How should one compile this? yarn ts-node doesn't work for me.

Apparently, you need to pass `--files` to `ts-node` for it to properly read the `tsconfig.json`:

> ```
> --files                         Load `files`, `include` and `exclude` from `tsconfig.json` on startup
> ```

If you just want to compile, `yarn tsc` does the job.

**Comment by baltpeter** [comment]
> Please add comments explaining the main script

Which parts do you think aren't clear enough without comments?

**Review by mal-tee — APPROVED** [review]
Thank you!

**Pull request merged**